src/subhalo_tracking.cpp

```diff
@@ -175,6 +176,151 @@ inline HBTInt GetLocalHostId(HBTInt pid, const HaloSnapshot_t &halo_snap, const
   }
   return hostid;
 }
+
+/* Identify and store the particle IDs of the most bound MinNumTracerPartOfSub
+ * collisionless tracers. Used to identify host FOF groups. */
+void GetTracerIds(vector<HBTInt>::iterator particle_ids, const Subhalo_t &Subhalo)
+{
+  /* Initialise vector. This will make it so the code knows when to stop looking
+   * for tracers, since orphans will have all but the first with NullParticleId */
+  fill(particle_ids, particle_ids + HBTConfig.MinNumTracerPartOfSub, SpecialConst::NullParticleId);
+
+  /* Iterate over the particle list to find tracers. */
+  int BoundRanking = 0;
+  for (auto particle = Subhalo.Particles.begin(); particle != Subhalo.Particles.begin() + Subhalo.Nbound; particle++)
+  {
+    if (particle->IsTracer()) // Only use tracers.
+      particle_ids[BoundRanking++] = particle->Id;
+
+    if (BoundRanking == HBTConfig.MinNumTracerPartOfSub) // Use up to a user-defined number of tracers.
```

Comment:
Do we want to create another variable that handles the number of tracers to use for host identification? Doing so would allow us to change to any value less or equal than MinNumTracerPartOfSub (including original behaviour of 1). We can make it default to MinNumTracerPartOfSub if unspecified.